resources/building-mobile-extensions/dispatching-events-from-your-extension.md

```diff
@@ -65,29 +45,45 @@ Event newEvent = new Event.Builder("MyCustomEvent",
                                    .setEventData(eventData).build();
 ```
 
-### Dispatch your Event
-
-After creating your event, dispatch it by using the `ACPCore` \(iOS\) / `MobileCore` \(Android\) method `dispatchEvent`. A typical place to dispatch an event is in an event listener.
+{% endtab %}
 
-In some cases, you may need to dispatch an event from one of your public APIs or application methods in order to trigger an internal flow in your own extension or in other Adobe extension.
+{% tab title="Objective-C" %}
 
 #### iOS
 
-```text
-#import "ACPCore_iOS/ACPExtensionEvent.h"
-
-- (IBAction) loginButtonClicked:(id)sender {
-    // construct the event to dispatch (see above)
-    ACPExtensionEvent* newEvent = ...;
+To create events in iOS, you must first import `ACPExtensionEvent.h` from the `ACPCore` framework:
 
-    // dispatch the event
-    NSError* error = nil;
-    if (![ACPCore dispatchEvent:newEvent error:&error] && error) {
-        NSLog(@"Error dispatching event %@:%ld", [error domain], [error code]);
-    };
+```objective-c
+#import "ACPCore_iOS/ACPExtensionEvent.h"
+...
+NSError* error = nil;
+NSDictionary* eventData = @{
+            @"id" : @"clickid53",
+            @"localTemp": @72.6,
+            @"pageClicks" : @23,
+            @"pageOrder" : @[@1,@2,@3,@4]
+};
+ACPExtensionEvent* newEvent = [ACPExtensionEvent extensionEventWithName:@"MyCustomEvent"          type:@"com.myCompany.eventType.custom"
+        source:@"com.myCompany.eventSource.custom" 
+        data:eventData
+        error:&error];
+if (error) {
+    NSLog(@"Error constructing new event %@:%ld", [error domain], [error code]);
 }
 ```
 
+{% endtab %}
+{% endtabs %}
+
+### Dispatch your Event
+
+After creating your event, dispatch it by using the `ACPCore` \(iOS\) / `MobileCore` \(Android\) method `dispatchEvent`. A typical place to dispatch an event is in an event listener.
+
+In some cases, you may need to dispatch an event from one of your public APIs or application methods in order to trigger an internal flow in your own extension or in other Adobe extension.
```

Comment:
Please make the following changes:

- "may" to "might" 
- "in order to" to "to" 
- from "in your own extension" to "your extension"

So that it reads as follows:

"In some cases, you might need to dispatch an event from one of your public APIs or application methods to trigger an internal flow in your extension or another Adobe extension."